### PR TITLE
ci: Remove caching for UI tests

### PR DESCRIPTION
**Pull request by philipphofmann**
The UI tests were using caches that stopped the UI tests from recompiling when there were changes in the library code. Instead, we need to recompile the test sample apps with their UI tests when there are changes in the library code to be able to catch bugs. This is fixed now by removing the cache.

Furthermore, @kevinrenskers opened a PR https://github.com/getsentry/sentry-cocoa/pull/2184 to enable automatic code signing for development of the iOS-Swift sample app. CI was green, and we merged it. Without caching we would have caught this error.

#skip-changelog

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -73,12 +73,17 @@ platform :ios do
...
+      readonly: true,
+      app_identifier: ["io.sentry.sample.iOS-Swift",  "io.sentry.sample.iOS-Swift.Clip"]
+    )
+
     build_app(
       workspace: "Sentry.xcworkspace",
       scheme: "iOS-Swift",
       derived_data_path: "DerivedData",
-      skip_archive: true,
-      skip_codesigning: true
```
Comment (kevinrenskers) [inline fastlane/Fastfile:81]: This seems to undo the fix from my PR

Comment (kevinrenskers) [reply fastlane/Fastfile:81]: Oh it was still a draft PR 😅

Comment (philipphofmann) [reply fastlane/Fastfile:81]: Yeah, I'm just playing araound.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**